code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm

```diff
@@ -0,0 +1,449 @@
+
+	sac_target.AdjustParalyzed(SACRIFICE_SLEEP_DURATION * 1.2)
+	sac_target.AdjustImmobilized(SACRIFICE_SLEEP_DURATION * 1.2)
+
+	addtimer(CALLBACK(src, .proc/after_target_sleeps, sac_target, destination), SACRIFICE_SLEEP_DURATION * 0.5) // Teleport to the minigame
+
+	return TRUE
+
+/**
+ * This proc is called from [proc/begin_sacrifice] after the [sac_target] falls asleep, shortly after the sacrifice occurs.
+ *
+ * Teleports the [sac_target] to the heretic room, asleep.
+ * If it fails to teleport, they will be disemboweled and stop the chain.
+ *
+ * Arguments
+ * * sac_target - the mob being sacrificed.
+ * * destination - the spot they're being teleported to.
+ */
+/datum/heretic_knowledge/hunt_and_sacrifice/proc/after_target_sleeps(mob/living/carbon/human/sac_target, turf/destination)
+	if(QDELETED(sac_target))
+		return
+
+	// Send 'em to the destination. If the teleport fails, just disembowel them and stop the chain
+	if(!destination || !do_teleport(sac_target, destination, asoundin = 'sound/magic/repulse.ogg', asoundout = 'sound/magic/blind.ogg', no_effects = TRUE, channel = TELEPORT_CHANNEL_MAGIC, forced = TRUE))
+		disembowel_target()
+		return
+
+	// If our target died during the (short) timer, and we fail to revive them, just disembowel them and stop the chain
+	if(!sac_target.heal_and_revive(75, span_danger("[sac_target]'s heart begins to beat with an unholy force as they return from death!")))
+		disembowel_target(sac_target)
+		return
+
+	to_chat(sac_target, span_big(span_hypnophrase("Unnatural forces begin to claw at your every being from beyond the veil.")))
+
+	sac_target.apply_status_effect(/datum/status_effect/unholy_determination, SACRIFICE_REALM_DURATION)
+	addtimer(CALLBACK(src, .proc/after_target_wakes, sac_target), SACRIFICE_SLEEP_DURATION * 0.5) // Begin the minigame
+
+	RegisterSignal(sac_target, COMSIG_MOVABLE_Z_CHANGED, .proc/on_target_escape) // Cheese condition
+	RegisterSignal(sac_target, COMSIG_LIVING_DEATH, .proc/on_target_death) // Loss condition
+
+/**
+ * This proc is called from [proc/after_target_sleeps] when the [sac_target] should be waking up.
+ *
+ * Begins the survival minigame, featuring the sacrifice targets.
+ * Gives them Helgrasp, throwing cursed hands towards them that they must dodge to survive.
+ * Also gives them a status effect, Unholy Determination, to help them in this endeavor.
+ *
+ * Then applies some miscellaneous effects.
+ */
+/datum/heretic_knowledge/hunt_and_sacrifice/proc/after_target_wakes(mob/living/carbon/human/sac_target)
+	if(QDELETED(sac_target))
+		return
+
+	// About how long should the helgrasp last? (1 metab a tick = helgrasp_time / 2 ticks (so, 1 minute = 60 seconds = 30 ticks))
+	var/helgrasp_time = 1 MINUTES
+
+	sac_target.reagents?.add_reagent(/datum/reagent/inverse/helgrasp, helgrasp_time / 20)
+	sac_target.apply_necropolis_curse(CURSE_BLINDING | CURSE_GRASPING)
```

Comment:
what if the target has increased metabolism (say, a virus), chem purge (virus or chemicals in their bloodstream) or no metabolism at all (skeletons)

Reply:
or if they already have helgrasp in their system?

Reply:
I do want to find an alternative to this to not rely on the reagent, 

The necropolis curse also spawns cursed hands, just not the same variety as helgrasp. So if they can't metab or purge the chem, they'll still be slapped around.